script/single-deployment/DeployAAVEBorrower.s.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.28;
+
+import {Script, console} from "forge-std/Script.sol";
+
+import {AaveBorrower} from "../../src/AaveBorrower.sol";
+import {FlashLoanRouter} from "../../src/FlashLoanRouter.sol";
+
+import {Constants} from "script/libraries/Constants.sol";
+
+contract DeployAAVEBorrower is Script {
+    function run() public virtual {
+        deployAAVEBorrower(FlashLoanRouter(address(0)));
+    }
+
+    // Deploy AaveBorrower with optional router input or fallback to env
+    function deployAAVEBorrower(FlashLoanRouter router) internal returns (AaveBorrower borrower) {
+        // Ensure the router address is provided or fallback to the environment variable
+        address routerAddress;
+        
+        if (address(router) != address(0)) {
+            routerAddress = address(router);
+        } else {
+            routerAddress = vm.envAddress("FLASHLOAN_ROUTER_ADDRESS");
+        }
```

Comment:
I wonder if instead of using this if statement (and having another env variable) we could simply expose a function on the router contract that returns its address given the current code/salt (ie. by instantiating it with `vm.startBroadcast`). Than each borrower would take this address as a mandatory argument.

We should never try to instantiate a borrower with a one off router contract address (otherwise the whole deployment script would become inconsistent).

Reply:
That's a good idea. 
Added a minor update in the latest commit for this.